### PR TITLE
[pyspark] Improve tutorial on enabling GPU support.

### PR DESCRIPTION
**Pull request by trivialfis**
- Quote the databricks doc on how to manage dependencies.
- Some wording changes.

### CODE REVIEW

File: doc/tutorials/spark_estimator.rst [hunk 1]
```diff
@@ -83,17 +83,50 @@ generate result dataset with 3 new columns:
...
+for more details on PySpark dependency management.
+
+In short, to create a Python environment that can be sent to a remote cluster using
+virtualenv and pip:
+
+.. code-block:: bash
+
+  python -m venv xgboost_env
+  source xgboost_env/bin/activate
+  pip install pyarrow pandas venv-pack cudf xgboost
```
Comment (mattf) [inline doc/tutorials/spark_estimator.rst:109]: cudf is installed w/ `pip install --extra-index-url=https://pypi.ngc.nvidia.com cudf-cu11`

Comment (trivialfis) [reply doc/tutorials/spark_estimator.rst]: Thank you for the review. Fixed along with a link to the install page.

File: doc/tutorials/spark_estimator.rst [hunk 2]
```diff
@@ -83,17 +83,52 @@ generate result dataset with 3 new columns:
 XGBoost PySpark GPU support
 ***************************
 
-XGBoost PySpark supports GPU training and prediction. To enable GPU support, first you
-need to install the XGBoost and the `cuDF <https://docs.rapids.ai/api/cudf/stable/>`_
-package. Then you can set `use_gpu` parameter to `True`.
+XGBoost PySpark fully supports GPU acceleration, users are not only able to enable
```
Comment (hcho3) [inline doc/tutorials/spark_estimator.rst:86]:
```suggestion
XGBoost PySpark fully supports GPU acceleration. Users are not only able to enable
```

File: doc/tutorials/spark_estimator.rst [hunk 3]
```diff
@@ -83,17 +83,52 @@ generate result dataset with 3 new columns:
...
+.. code-block:: bash
+
+  python -m venv xgboost_env
+  source xgboost_env/bin/activate
+  pip install pyarrow pandas venv-pack xgboost
+  # https://rapids.ai/pip.html#install
+  pip install cudf-cu11 --extra-index-url=https://pypi.ngc.nvidia.com
+  venv-pack -o xgboost_env.tar.gz
+
+with conda:
```
Comment (hcho3) [inline doc/tutorials/spark_estimator.rst:114]:
```suggestion
With Conda:
```

**Review by hcho3 — CHANGES_REQUESTED** [review]

**Review by hcho3 — APPROVED** [review]

**Pull request merged**